### DISCUSSION TITLE
No squash rule for PRs

### DISCUSSION THREAD

**Discussion by connorhu** [FriendsOfSymfony1/symfony1#356] [Ideas]
It occurred to me in one of the PRs that it's hard to make sense of a PR when you're close to a merge if we're always doing squash. My suggestion would be to never squash a PR by the owner, only squash it on merge. That way it might remain more understandable what the evolution of the development was, what and why a particular commit changes in the PR.
Certain things can be squashed, but the logic of the evolution should be traceable.

**Comment by alquerci** [comment]
As of an example on this PR #301

Is it OK to use `git commit --fixup=COMMIT_HASH`?

I like to commit each time the test passes and more.

Ideally `test/bin/test && git commit -p --fixup=COMMIT_HASH`
Sadly, the full test suite is too slow, that's another issue.
